src/seamless-immutable.js

```diff
@@ -471,8 +471,17 @@
     return makeImmutable(obj, mutatingObjectMethods);
   }
 
+  var REACT_ELEMENT_TYPE = typeof Symbol === 'function' && Symbol.for && Symbol.for('react.element');
+  var REACT_ELEMENT_TYPE_FALLBACK = 0xeac7;
+
+  function isReactElement(obj) {
+    return obj.hasOwnProperty &&
+           obj.hasOwnProperty('$$typeof') &&
+           obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE;
+  }
```

Comment:
I think it should be okay in IE11. I've kept your ```…_TYPE_FALLBACK || …_TYPE``` condition intact.

The only difference between your `isReactElement` and `React.isValidElement` is that you test for `hasOwnProperty`, while they test for `typeof obj` and `obj !== null`.

> Do you think I should add a dev dependency on React so I can test the detection?

Personally I don't see any problems with it. But take it with a grain of salt - I'm not a maintainer of this repo, @rtfeldman is.